flycheck.el

```diff
@@ -9480,6 +9480,26 @@ or if the current buffer has no file name."
   (and buffer-file-name
        (locate-dominating-file buffer-file-name "Cargo.toml")))
 
+(defun flycheck-rust-cargo-metadata ()
+  "Exec 'cargo metadata' and return the result as parsed JSON object.
+
+Return nil if the command failed or if a parse error occured."
+  (let ((metadata-lines
+         (ignore-errors
+           (process-lines "cargo" "metadata" "--no-deps" "--format-version" "1"))))
+    (and metadata-lines
+         (ignore-errors
+           (json-read-from-string (string-join metadata-lines ""))))))
```

Comment:
That call to string-join should use `\n`, not ``, I think (though it might not matter in this particular case?)
Besides, would it read nicer to use `call-process` and `json-read` instead of `process-lines` + `string-join` + `json-read-from-string`?

Reply:
In this case, we may just use `flycheck-parse-json`?

Reply:
@fmdkdd 
`flycheck-parse-json` returns a list of parsed JSON lines, so isn't it too complex for this case, where what we want is only `metadata` ?

Reply:
It may be overkill, yes.  Using `call-process` and `json-read` directly is fine :)